launcher/updatedialog_moc.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+ * updatedialog_moc.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#include "StdInc.h"
+#include "updatedialog_moc.h"
+#include "ui_updatedialog_moc.h"
+
+#include "../lib/CConfigHandler.h"
+#include "../lib/GameConstants.h"
+
+#include <QNetworkAccessManager>
+#include <QNetworkReply>
+#include <QNetworkRequest>
+
+UpdateDialog::UpdateDialog(QWidget *parent, bool calledManually) :
+	QDialog(parent),
+	ui(new Ui::UpdateDialog),
+	calledManually(calledManually)
+{
+	ui->setupUi(this);
+	
+	if(calledManually)
+	{
+		setWindowModality(Qt::ApplicationModal);
+		show();
+	}
+	
+	connect(ui->closeButton, SIGNAL(clicked()), this, SLOT(close()));
+	
+	if(settings["launcher"]["updateOnStartup"].Bool() == true)
+		ui->checkOnStartup->setCheckState(Qt::CheckState::Checked);
+	
+	currentVersion = GameConstants::VCMI_VERSION;
+	
+	setWindowTitle(QString::fromStdString(currentVersion));
+	
+#ifdef VCMI_WINDOWS
+	platformParameter = "windows";
+#elif defined(VCMI_MAC)
+	platformParameter = "macos";
+#elif defined(VCMI_IOS)
+	platformParameter = "ios";
+#elif defined(VCMI_ANDROID)
+	platformParameter = "android";
+#elif defined(VCMI_UNIX)
+	platformParameter = "linux";
+#endif
+	
+	QString url = QString::fromStdString(settings["launcher"]["updateConfigUrl"].String());
+	
+	QScopedPointer<QNetworkAccessManager> manager(new QNetworkAccessManager);
```

Comment:
I wouldn't call it a standard pattern. Why not simply deleting the manager inside the response handler (which you already do)?

Reply:
I'd say that the simplest and safest way would be making the network manager a private field and not a pointer. Or even simply:

```c++
auto manager = new QNetworkAccessManager{this};
```

and remove `response->manager()->deleteLater();`